format-specs/parquet-raster.md

```diff
@@ -0,0 +1,122 @@
+| `isNodataValue`  | 1 bit          | If true, all values of the band are expected to be nodata values. This is a dirty flag; to set it properly, the function `st_bandisnodata` must be called with `TRUE` as the last argument.                                                                                                                                  |
+| `isGZIPPed`      | 1 bit          | If true, the data is compressed using GZIP before being passed to the Parquet compression process.                                                                                                                                                                                                                            |
+| `pixtype`        | 4 bits         | Pixel type: <br>0: 1-bit boolean<br>1: 2-bit unsigned integer<br>2: 4-bit unsigned integer<br>3: 8-bit signed integer<br>4: 8-bit unsigned integer<br>5: 16-bit signed integer<br>6: 16-bit unsigned signed integer<br>7: 32-bit signed integer<br>8: 32-bit unsigned signed integer<br>10: 32-bit float<br>11: 64-bit float |
+| `nodata`         | 1 to 8 bytes (depending on `pixtype` [1]) | Nodata value.                                                                                                                                                                                                                                                                                                                |
+| `data`           | byte_array   | Raster band pixel data (see below).                                                                                                                                                                                                                                                                                          |
+
+### In-DB pixel data encoding
+
+This encoding is used when `isOffline` flag is false.
+
+| Name         | Type            | Meaning |
+|--------------|-----------------|---------|
+| `pix[w*h]`   | 1 to 8 bytes (depending on `pixtype` [1]) | Pixel values, row after row. `pix[0]` is the upper-left, `pix[w-1]` is the upper-right. <br><br>Endianness is specified at the start of WKB. It is implicit up to 8 bits (bit-order is most significant first). |
+
+### Out-DB pixel data encoding
+
+This encoding is used when `isOffline` flag is true.
+
+| Name         | Type      | Meaning                                                                 |
+|--------------|-----------|-------------------------------------------------------------------------|
+| `bandNumber` | int8       | 0-based band number to use from the set available in the external file. |
+| `url`        | string     | The URI of the out-db raster file (e.g., GeoTIFF files)                 |
```

Comment:
I think we should flesh out 'e.g., GeoTIFF files' a lot more. What types of raster data do we expect clients to support? What happens when someone puts in a value that the client doesn't support? Do we expect them to figure out if they can read it from the file name? Or are they expected to try to open it? 

In STAC we don't just use a single url field, we have an object that also has a  'media type', see https://github.com/radiantearth/stac-spec/blob/master/best-practices.md#working-with-media-types for info on how we use it. Clients can make use of that to figure out if they support it. We don't try to list all potential media types, but we share the common ones. An approach like that could potentiallyw ork here, but we'd need another field.